src/main/java/com/webonise/tomcat8/redisession/RedisSessionManager.java

```diff
@@ -16,20 +22,39 @@
 import java.util.function.*;
 import java.util.stream.*;
 
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+
+
+
+
+
+
 /**
  * Responsible for creating sessions that persist into Redis.
  */
-public class RedisSessionManager implements Manager {
+public class RedisSessionManager extends LifecycleMBeanBase implements Manager {
 
   private static final Log LOG = LogFactory.getLog(RedisSessionManager.class);
   private final AtomicLong sessionCounter = new AtomicLong(0L);
   private final PropertyChangeSupport changeListeners = new PropertyChangeSupport(this);
-  private final Redis redis = new Redis(); // TODO Allow this to be populated from XML config
+  private  Redis redis ; // TODO Allow this to be populated from XML config
```

Comment:
Get rid of the TODO, but make this `volatile`. Every property should either be `volatile` or `final`, or you're leaving yourself open to really weird threading bugs.